### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by SSubhash05**
i have made the changes mentioned in the comments, please verify

### CODE REVIEW

**Review by sathisb — DISMISSED** [review]

**Review by ramsonk — DISMISSED** [review]

File: Employee_details_using _structures/EmployeeDetails.c [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+{
+	int string_length=strlen(employee_name),indexOfchar=0;
+	if(string_length>NAME_LENGTH)
+	{
+		printf("more than 10 characters!!\n");
+		return false;
+	}
+	while(indexOfchar<string_length)
+		{
+			if((employee_name[indexOfchar]>='A' && employee_name[indexOfchar]<='Z')||( employee_name[indexOfchar]>='a' && employee_name[indexOfchar]<='z'))
```
Comment (sathisb) [inline Employee_details_using _structures/EmployeeDetails.c]: What should be the maximum number of characters in a single line

File: Employee_details_using_linkedlists/linkedlist.c [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+	} while(!stringValidation(employee_ptr->employee_designation));
+
+	if(!first_employee)
+	{
+		employee_ptr->next_employee=NULL;
+		first_employee=employee_ptr;
+	}
+	else
+	{
+		printf("2nd node\n");
```
Comment (ramsonk) [inline Employee_details_using_linkedlists/linkedlist.c]: remove all debug prints

**Comment by sathisb** [comment]
1. Remove all the unnecessary debug prints in the code.
2. Add necessary error prints user should know.
3. Name inputs should be accept with space.

File: Employee_details_using _structures/EmployeeDetails.c [hunk 3]
```diff
@@ -93,85 +99,75 @@ struct employee* add_employee(struct employee *employee_ptr,int *employee_count)
...
-
+	printf("size of employee =%d\n",sizeof(struct employee));
+	printf("size of employee ptr =%d\n",sizeof(employee_ptr));
+	// for(index=0;index<employee_count;index++)
+	// {
+	//  	printf("NAME        :%s\n", employee_ptr[index].employee_name);
+	//  	printf("ID          :%d\n", employee_ptr[index].employee_id);
+	//  	printf("AGE         :%d\n", employee_ptr[index].employee_age);
+	//  	printf("DESIGNATION :%s\n\n", employee_ptr[index].employee_designation);
+  // }
```
Comment (ramsonk) [inline Employee_details_using _structures/EmployeeDetails.c]: already told to remove debug prints and why these lines are commented, then what will you display?

File: Employee_details_using _structures/EmployeeDetails.c [hunk 4]
```diff
@@ -235,59 +229,58 @@ bool remove_employee_byname(struct employee *employee_ptr,int *employee_count) /
...
+						index++;
+					}
+	        index++;
 				}
-        index++;
-			}
 
+			}
+		}
+		if(name_check)
```
Comment (ramsonk) [inline Employee_details_using _structures/EmployeeDetails.c:249]: Think of how you remove this variable and this if else condition

File: Employee_details_using_linkedlists/linkedlist.c [hunk 5]
```diff
@@ -119,34 +123,37 @@ void display_employee_details(struct employee* first_employee)
...
 	else
 	{
-		printf("add any employee to view their details\n");
+		printf("\nadd any employee to view their details\n");
 	}
 }
 
-struct employee* removeEmployee(struct employee * first_employee,int sorting_type)
+struct employee* removeEmployee(struct employee * first_employee,
+	                              int sorting_type)
```
Comment (ramsonk) [inline Employee_details_using_linkedlists/linkedlist.c:131]: In remove function, sorting_type variable name is used, is that relevant??

File: avl_tree/avl_tree.c [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        root=insertNode(root,value);
+        break;
+      case 2:
+        // if(root->right_ptr->left_ptr==NULL && root->right_ptr->right_ptr==NULL)
+        // {
+        // printf("root value :%d",root->data);
+        // }
+        // printf("Enter the value to be removed =");
+        // //scanf("%d",&value);
+        //root=insertNode(root,value);
```
Comment (ramsonk) [inline avl_tree/avl_tree.c]: why a commented piece of code is present ?

**Review by ramsonk — CHANGES_REQUESTED** [review]
make changes mentioned in the comments

File: Employee_details_using _structures/EmployeeDetails.c (hunk 1 above)
Comment (SSubhash05) [reply Employee_details_using _structures/EmployeeDetails.c]: Maximum 78 characters per line.I have changed it in my recent commit.

File: Employee_details_using_linkedlists/linkedlist.c (hunk 2 above)
Comment (SSubhash05) [reply Employee_details_using_linkedlists/linkedlist.c]: Yes I removed  everything in the recent commit.

File: Employee_details_using _structures/EmployeeDetails.c (hunk 3 above)
Comment (SSubhash05) [reply Employee_details_using _structures/EmployeeDetails.c]: I have wrongly committed and changed it in recent commit.

**Review by ramsonk — APPROVED** [review]

**Review by Barath2497 — APPROVED** [review]

**Review by sathisb — APPROVED** [review]

**Pull request merged**